client/src/components/SignUp.js

```diff
@@ -1,41 +1,102 @@
-import React, { Component } from "react";
+import React, {Component} from "react";
+import {Alert} from 'reactstrap';
+import {post} from "../helper/Fetch";
+import {ReroutToLogin} from "./ReroutToLogin";
+
 export default class SignUp extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            email: "",
+            password: "",
+            firstName: "",
+            lastName: "",
+            username: "",
+            reloginPrompt: false,
+            redirect: false,
+        };
+    }
+
+    handleFieldChange = (e, key) => {
+        this.setState({
+            [key]: e.target.value,
+
+        });
+    }
+
+    handleLogin = (e) => {
+        e.preventDefault(); // Prevents the form from submitting when Submit button is clicked
+        post("api/v1/signUp", {
+                email: this.state.email,
```

Comment:
Same as above. It doesn't need the email field (the email and username fields are the same).